pages/api/property/details.js

```diff
@@ -0,0 +1,61 @@
+// details api gets the sam_id passed.
+// then grabs the data related to sam_id
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
+export default async function handler(req, res) {
+  try {
+    const sam_id = req.query.sam_id;
+
+    if (!sam_id) {
+        return res.status(400).json({ error: 'SAM_ID is required.' });
+    }
+
+    const samData = await prisma.sam.findFirst({
+        where: {
+            SAM_ADDRESS_ID: sam_id,
+        },
+        select: {
+            FULL_ADDRESS: true,
+            MAILING_NEIGHBORHOOD: true,
+            ZIP_CODE: true,
+        },
+    });
+  
+    if (!samData) {
+        return res.status(404).json({ error: 'SAM record not found.' });
+    }
+  
+    const propertyData = await prisma.property.findFirst({
+        where: {
+            PID: samData.PARCEL,
+        },
+        select: {
+            OWNER: true
+        }
+    });
+  
+    const bpvData = await prisma.bpv.findFirst({
+        where: {
+            sam_id: sam_id,
+        },
+        select: {
+            status: true,
+            code: true,
+            description: true
+        }
+    });
```

Comment:
Does this need to be restricted to the codes we care about? Or are the only codes in the DB the ones we care about?

Reply:
This needs restriction over in server side. I will add a implementation of restriction code and comment it out for now. Some of the restriction code Wayne talked about doesn't exist. I think Wayne has to look at the code types we have(this is inside the badlandlords thread) and give use which code to choose as Scofflaw. This is different format than what Wayne gave like CMR 410 and CRM 410. What he gave doesn't exist in the actual database.